lib/server/services/qualityControlFlag/QcFlagTypeService.js

```diff
@@ -82,26 +81,28 @@ class QcFlagTypeService {
         } = parameters;
 
         const user = await getUserOrFail({ userId, externalUserId });
-        if (color && !/#[0-9a-fA-F]{6}/.test(color)) {
-            throw new BadParameterError(`Incorrect format of the color provided (${color})`);
-        }
 
-        try {
-            const newFlagInstance = await QcFlagTypeRepository.insert({
-                name,
-                method,
-                bad,
-                color: color ?? (bad ? DEFAULT_BAD_FLAG_COLOR : DEFAULT_NOT_BAD_FLAG_COLOR),
-                createdById: user.id,
-            });
-            return qcFlagTypeAdapter.toEntity(await this.getById(newFlagInstance.id));
-        } catch (sequelizeError) {
-            if (sequelizeError instanceof sequelize.ValidationError) {
-                throw new ConflictError(sequelizeError.errors.map(({ message }) => message).join(';'));
-            } else {
-                throw sequelizeError;
-            }
+        let conflictingEntity = await QcFlagTypeRepository.findOne({
```

Comment:
I would rather name it `existingQcFlagType` no?

Reply:
renamed